### ISSUE TITLE
unable to install after a new release; assets not attach to release

### ISSUE THREAD

**Issue by ulfbruhn** [gruntwork-io/terragrunt#2847]
**Describe the bug**
After the new release was triggered, github had the new release listed as "latest" but the assets e.g. "terragrunt_linux_amd64" were not present in the release. as a result being unable to install the latest version of terragrunt.
13min between release publish and last asset added.

**To Reproduce**

**Expected behavior**
Being able to use the new release as soon as its published.
Publish release only after all assets have finished building and are attached to the release.

**Nice to have**
![image](https://github.com/gruntwork-io/terragrunt/assets/129742869/90db9d98-26a0-4ce3-ae0f-664a82ffce1b)

**Versions**
- Terragrunt version: 0.54.2

**Comment by denis256** [comment]
It takes a while to run all CICD jobs and publish release... :hourglass_flowing_sand:

**Comment by ulfbruhn** [comment]
Well, but this led to a time frame of around 15min where the tool was unusable in an automated deployment pipeline that automatically installs the newest version. why publish the release before all CICD jobs are finished and all assets are attached to the release.
